src/functionalTest/kotlin/com/github/jengelman/gradle/plugins/shadow/FindResourceInClasspathTest.kt

```diff
@@ -0,0 +1,63 @@
+package com.github.jengelman.gradle.plugins.shadow
+
+import assertk.assertThat
+import assertk.assertions.contains
+import assertk.assertions.doesNotContain
+import com.github.jengelman.gradle.plugins.shadow.testkit.containsOnly
+import kotlin.io.path.appendText
+import org.junit.jupiter.api.Test
+
+class FindResourceInClasspathTest : BasePluginTest() {
+  @Test
+  fun findResourceInClasspath() {
+    projectScript.appendText(
+      """
+        dependencies {
+          implementation 'my:a:1.0'
+        }
+        tasks.register('find1', com.github.jengelman.gradle.plugins.shadow.tasks.FindResourceInClasspath) {
+          classpath = configurations.runtimeClasspath
+        }
+        tasks.register('find2', com.github.jengelman.gradle.plugins.shadow.tasks.FindResourceInClasspath) {
+          classpath = configurations.runtimeClasspath
+          include("a.properties")
+        }
+        tasks.register('find3', com.github.jengelman.gradle.plugins.shadow.tasks.FindResourceInClasspath) {
+          classpath = configurations.runtimeClasspath
+          exclude("a.properties")
+        }
+      """.trimIndent(),
+    )
+
+    val result1 = runWithSuccess(":find1")
+    assertThat(result1.output).contains(
+      "> Task :find1",
+      "scanning ",
+      "/my/a/1.0/a-1.0.jar",
+      "/a.properties",
+      "/a2.properties",
+    )
+
+    val result2 = runWithSuccess(":find2")
+    assertThat(result2.output).contains(
+      "> Task :find2",
+      "scanning ",
+      "/my/a/1.0/a-1.0.jar",
+      "/a.properties",
+    )
+    assertThat(result2.output).doesNotContain(
```

Comment:
You can use `all` here

```kt
  assertThat(result2.output).all {
    contains(...)
    doesNotContain(...)
  }
```

BTW, these messages could be checked by regex by `containsMatch`.

Reply:
It would greater to match the output format you want:

```
> Task :find1
scanning /var/folders/ry/cppncsdj3rd634f18p7m2n000000gn/T/10772270132859729920/local-maven-repo/my/a/1.0/a-1.0.jar
  -> /private/var/folders/ry/cppncsdj3rd634f18p7m2n000000gn/T/junit-3825987517199813433/build/tmp/.cache/expanded/zip_2dda8f61e388988b7bdf54222a9de8c8/a.properties
  -> /private/var/folders/ry/cppncsdj3rd634f18p7m2n000000gn/T/junit-3825987517199813433/build/tmp/.cache/expanded/zip_2dda8f61e388988b7bdf54222a9de8c8/a2.properties
```